File: easy/word_to_digit.pl
use strict;
my %replace = (
    'zero'  => 0,
    'one'   => 1,
    'two'   => 2,
    'three' => 3,
    'four'  => 4,
    'five'  => 5,
    'six'   => 6,
    'seven' => 7,
    'eight' => 8,
    'nine'  => 9,
    ';'     => ''
);
my $words = qr/@{[join '|', map { quotemeta($_) } keys %replace]}/;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    next if ( $line =~ m/^\s$/ );
    chomp($line);
    $line =~ s/($words)/$replace{$1}/g;
    print "$line\n";
}
close(INFILE);

File: easy/word_to_digit.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strings"
)

var digits map[string]string

func init() {
	digits = map[string]string{"zero": "0", "one": "1", "two": "2",
		"three": "3", "four": "4", "five": "5", "six": "6",
		"seven": "7", "eight": "8", "nine": "9"}
}

func wordToDigit(q string) string {
	t := strings.Split(q, ";")
	r := make([]string, len(t))
	for ix, i := range t {
		r[ix] = digits[i]
	}
	return strings.Join(r, "")
}

func main() {
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	defer data.Close()

	scanner := bufio.NewScanner(data)
	for scanner.Scan() {
		fmt.Println(wordToDigit(scanner.Text()))
	}
}
